diesel_codegen/src/schema_inference.rs

```diff
@@ -53,6 +50,31 @@ pub fn derive_infer_table_from_schema(input: syn::MacroInput) -> quote::Tokens {
     })
 }
 
+fn infer_schema_for_schema_name(database_url: &str, schema_name: Option<&str>) -> quote::Tokens {
+    let table_names = load_table_names(&database_url, schema_name).unwrap();
+    let schema_inferences = table_names.into_iter().map(|table_name| {
+        let mod_ident = syn::Ident::new(format!("infer_{}", table_name));
+        let table_name = match schema_name {
+            Some(name) => format!("{}.{}", name, table_name),
+            None => table_name,
+        };
+        quote! {
+            mod #mod_ident {
+                infer_table_from_schema!(#database_url, #table_name);
+            }
+            pub use self::#mod_ident::*;
+        }
+    });
+
+    match schema_name {
+        Some(name) => {
+            let schema_ident = syn::Ident::new(name);
+            quote! { pub mod #schema_ident { #(schema_inferences)* } }
+        }
+        None => quote!(#(schema_inferences)*),
+    }
```

Comment:
Added it to the docs. I don't think we need to specifically call out the error case there, as I don't think having a table with the same name as a schema is a common thing to do, especially for tables in `public`. If the error did occur I think it's pretty self-explanatory what happened.